### ISSUE TITLE
Next episodes comes up empty

### ISSUE THREAD

**Issue by cgcgcgcgcg** [Ezra-Hubbard/Ezra-Hubbard.github.io#12]
An empty widget is a lonely widget.

**Comment by umbrellaplug** [comment]
Can you give more information on this issue? Are you using a skin? Can you give me the steps to re-produce the issue so that I can look into it?

**Comment by cgcgcgcgcg** [comment]
> Can you give more information on this issue? Are you using a skin? Can you give me the steps to re-produce the issue so that I can look into it?

Using Umbrella's Next episodes as a widget doesn't work. I tried Xonfluence and Titan MOD skins.
I have uninstalled it, but will install it again to check and let you know.

**Comment by cgcgcgcgcg** [comment]
TV Shows > Next Episodes folder is empty. So, can't really use that folder as a widget on any skin.
On an original Fen fixed by another dev to work, Next episodes works fine.

https://user-images.githubusercontent.com/38960370/176738451-91e32c3a-b2c1-4ba9-8a6b-52f6522f0c66.mp4

**Comment by cgcgcgcgcg** [comment]
I'm using your My Accounts with both Trakt and debrid Revoked, and reauthorized, and Ezra Resynced -also Fen, hence the Trakt error on the video.

**Comment by umbrellaplug** [comment]
You have something else going on. I would need to see the logs and what was coming in from trakt to better check it. Thank you for the videos, that helps tremendously. I went ahead and did the same for you. It's something with the results from trakt account or something else. Can you turn on debugging in logging in the app and set it to go to kodi.log then try it again and send me the logs from that?

https://user-images.githubusercontent.com/107957815/176743583-9ff56468-dcc2-45a8-9a4e-88bb6b5ab04d.mp4

**Comment by cgcgcgcgcg** [comment]
Actually after putting my API credentials -which is a pain in Linux since there's no copy & paste in Linux's Kodi-, and authorizing Trakt, Next Episodes in Ezra, gave an error, but episodes appeared -pics.
I then Revoked Trakt, used no API, reauthorized, and Next Episodes worked after the error again.
![k_20220630-204718](https://user-images.githubusercontent.com/38960370/176743756-e1dd650e-0fe7-4c2c-a006-1a7b774b8bee.png)
![k_20220630-204743](https://user-images.githubusercontent.com/38960370/176743761-6d01b389-35b5-44d8-8566-86bc4a8fc153.png)

**Comment by Ezra-Hubbard** [comment]
Yeh, I can't replicate here, its working as expected on my current install

**Comment by cgcgcgcgcg** [comment]
> You have something else going on. I would need to see the logs and what was coming in from trakt to better check it. Thank you for the videos, that helps tremendously. I went ahead and did the same for you. It's something with the results from trakt account or something else. Can you turn on debugging in logging in the app and set it to go to kodi.log then try it again and send me the logs from that?
> 
>  Untitled.mp4

It's my build's log so it's kinda big. Line 3147 looks interesting.
[kodi.log](https://github.com/Ezra-Hubbard/Ezra-Hubbard.github.io/files/9022189/kodi.log)

**Comment by cgcgcgcgcg** [comment]
Bear in mind that when I first encountered the Next Episodes bug, I was running a clean build (no fixed Venom & Fen), with Ezra's My Accounts.

**Comment by umbrellaplug** [comment]
> > You have something else going on. I would need to see the logs and what was coming in from trakt to better check it. Thank you for the videos, that helps tremendously. I went ahead and did the same for you. It's something with the results from trakt account or something else. Can you turn on debugging in logging in the app and set it to go to kodi.log then try it again and send me the logs from that?
> > Untitled.mp4
> 
> It's my build's log so it's kinda big. Line 3147 looks interesting. [kodi.log](https://github.com/Ezra-Hubbard/Ezra-Hubbard.github.io/files/9022189/kodi.log)

"Throws hands up and immediately stomps away after reading builds"..joking. I don't know what the hate for builds is about yet. Maybe I'll learn. :)

It looks like some confusion with trakt tokens. I'd be interested to know if your trakt.token is the same in the settings.xml between the my accounts module and ezra.

When you authorize trakt it sends client id and client secret. Trakt sends token back. We then use token on every request. If token gets expired or jacked up we can request new token using trakt.refresh which is a different token that was returned and originally stored when we authorized trakt, Original token is supposed to be good for 3 months, but things happen. This is why I ask if they are the same in settings.xml

**Comment by cgcgcgcgcg** [comment]
> > > You have something else going on. I would need to see the logs and what was coming in from trakt to better check it. Thank you for the videos, that helps tremendously. I went ahead and did the same for you. It's something with the results from trakt account or something else. Can you turn on debugging in logging in the app and set it to go to kodi.log then try it again and send me the logs from that?
> > > Untitled.mp4
> > 
> > 
> > It's my build's log so it's kinda big. Line 3147 looks interesting. [kodi.log](https://github.com/Ezra-Hubbard/Ezra-Hubbard.github.io/files/9022189/kodi.log)
> 
> "Throws hands up and immediately stomps away after reading builds"..joking. I don't know what the hate for builds is about yet. Maybe I'll learn. :)
> 
> It looks like some confusion with trakt tokens. I'd be interested to know if your trakt.token is the same in the settings.xml between the my accounts module and ezra.
> 
> When you authorize trakt it sends client id and client secret. Trakt sends token back. We then use token on every request. If token gets expired or jacked up we can request new token using trakt.refresh which is a different token that was returned and originally stored when we authorized trakt, Original token is supposed to be good for 3 months, but things happen. This is why I ask if they are the same in settings.xml

After revoking Trakt in My Accounts for the last time I didn't use an API, I clicked on it and was blank, so I thought "F#ck it" and just authorized Trakt as usual.

**Comment by cgcgcgcgcg** [comment]
OK, I thought what I didn't do before?
Rebooting.
Rebooted, Revoked Trakt, reauthorized, Resynced, and Next Episodes work fine, no errors.
Sorry for the trouble peeps.
![k_20220630-211205](https://user-images.githubusercontent.com/38960370/176748053-0a3a7ba0-7c8c-4a3d-a272-01066eaa01c3.png)

**Comment by umbrellaplug** [comment]
Nice! Mark another one closed Ezra. :)

Hopefully after this latest version of my accounts gets out and about to more people we will start seeing less and less of these issues with trakt, Fingers crossed. I know I'm about to start clearing some people's bad tokens for them. :)

**Comment by umbrellaplug** [comment]
I’m the umbrella dev, but this is the Ezra GitHub. The issues I was looking
at here we’re in relation to Ezra. Just trying to help the community as a
whole. Your response says using “Umbrella’s Next episodes as a widget”. I’m
hoping you meant Ezra. If this is an issue with my side I’m happy to look
into it but I would ask out of respect to Ezra that if you’re reporting an
issue with umbrella, please do that on the umbrella GitHub. We’re using
different clones. Ezra from Fen and Umbrella from Venom. Hopefully This all
makes sense. Sorry to be long winded, just want to make sure we’re not
confusing issues or crossing plugins on issue reporting.
Some of the issues being reported may effect both plugins though so I like
to look over the other devs issue reports.

**Comment by cg00001** [comment]
> I’m the umbrella dev, but this is the Ezra GitHub. The issues I was looking at here we’re in relation to Ezra. Just trying to help the community as a whole. 

I don't use Ezra any more, but Umbrella is my #1 addon. Using Progress works fine, of course peeps should make sure to hide shows they don't watch any more in trakt.tv > Progress, like this for example:
https://diggzwiki.com/index.php?title=Newbies#Clearing_Progress_in_Trakt
![k152447](https://user-images.githubusercontent.com/38960370/195090578-e0ebcd9d-a330-49c8-bd0d-29e72505aee1.png)

**Comment by ZacWolf** [comment]
I hope this isn't adding noise, but when I had a similar issue with Ezra not displaying my "Next Episode" list, I tracked it down to a Kodi database file version issue. 

I moved the following files:
```bash
cd ~/.kodi/userdata/addon_data/plugin.video.ezra/databases
mkdir old
mv traktcache4.db old
mv views.db old
mv watched.db old
reboot
```
After rebooting the machine, restarted Ezra and it started working again.

This happened a long while ago, but I seem to recall the issue appeared after upgrading Kodi from a pre-Leia version to a post-Leia version.
